clang-tools-extra/clang-tidy/hooks/GuardInForCheck.h

```diff
@@ -20,8 +20,22 @@ namespace hooks {
 /// #define GUARD(maxiter) _g(__LINE__, (maxiter)+1)
 /// for (int i = 0; GUARD(3), i < 3; ++i)
 ///
-/// This is the only way to satisfy the guard rule when using a
-/// for-loop in C.
+/// To satisfy the guard rule when using a for-loop in C guard should be 
+/// place either in the condition part of the loop, or as a first call in loop body, e.g.
+///
+/// for(int i = 0; i < 3; ++i) {
+///   GUARD(3);
+/// }
+/// 
+/// In case of nested loops, the guard limit value should be 
+/// multiplied by a number of iterations in each loop, e.g.
+///
+/// for(int i = 0; GUARD(3), i < 3; ++i) {
+///   for (int j = 0; GUARD(17), j < 5; ++j)
+/// }
+/// 
+/// (most descendant loop iterations + 1) * (each parent loops iterations) - 1
+///
 class GuardInForCheck : public ClangTidyCheck {
```

Comment:
I guess I'm still not clear about computing the maxiter suggestion for a nested loop from from an outer loop's call to `_g` - for example, 
```
    for (int i = 0; GUARD(2), i != 3; ++i) {
        if (i == 3) break;        
	     for (int j = 0; j < 4; ++j)
		    trace_num("j", 1, j);
    }
```
warns but offers no fix - shouldn't it? IMHO the documentation could be extended a bit... :-)

Reply:
I'll check it. Generally sometimes it's too complicated to calculate properly guard for nested loops and there is no hint displayed (as it might be -1 as in previous cases). So instead of displaying stupid hints it displays nothing :)
As for the documentation - you mean the one in header file?

Reply:
Also any `if` (or other) conditions between nested loops are not taken into account to calculate the guard limit. Calculation is strictly based on parent for conditions (init value, condition limit and increment) or on existing parent GUARD.

Reply:
> As for the documentation - you mean the one in header file?

yes

Reply:
> Also any `if` (or other) conditions between nested loops are not taken into account to calculate the guard limit. Calculation is strictly based on parent for conditions (init value, condition limit and increment) or on existing parent GUARD.

Parent guard in the for loop, i think; outer while loops are not checked for guards - right? Also, I think it should be documented that the explicit guard takes precedence - the check won't try to compute an alternative iteration limit if the programmer explicitly says what it should be, and if the programmer says it wrong, it's just their problem...

Reply:
Yes, parent guards in for loops only.